src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt

```diff
@@ -93,4 +108,19 @@ class OpenApiConfiguration(
       }
     }
   }
+
+  private fun errorSchema(): Schema<Any> {
+    val errorResponseSchema = Schema<Any>()
+    errorResponseSchema.name = "ErrorResponse"
+    errorResponseSchema.`$ref` = "#/components/schemas/ErrorResponse"
+    return errorResponseSchema
+  }
+
+  private fun createApiResponse(message: String, schema: Schema<Any>): ApiResponse {
+    return ApiResponse()
+      .description(message)
+      .content(
+        Content().addMediaType(MediaType.APPLICATION_JSON_VALUE, io.swagger.v3.oas.models.media.MediaType().schema(schema)),
+      )
+  }
```

Comment:
any reason to not take schema as an argument and just add the error schema in here, something like 
```suggestion
  private fun createErrorApiResponse(message: String): ApiResponse {
    val errorResponseSchema = Schema<Any>()
    errorResponseSchema.name = "ErrorResponse"
    errorResponseSchema.`$ref` = "#/components/schemas/ErrorResponse"
    return ApiResponse()
      .description(message)
      .content(
        Content().addMediaType(MediaType.APPLICATION_JSON_VALUE, io.swagger.v3.oas.models.media.MediaType().schema(errorResponseSchema)),
      )
  }
```